app/src/main/java/org/mozilla/focus/tiles/HomeTilesManager.kt

```diff
@@ -0,0 +1,116 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.tiles
+
+import android.content.Context
+import android.content.Context.MODE_PRIVATE
+import android.content.SharedPreferences
+import android.support.annotation.UiThread
+import org.json.JSONArray
+import org.json.JSONObject
+
+private const val HOME_TILES = "homeTiles"
+private const val CUSTOM_SITES_LIST = "customSitesList"
+
+private const val KEY_URL = "url"
+private const val KEY_TITLE = "title"
+private const val KEY_IMG = "img"
+private const val KEY_ID = "identifier"
+
+data class HomeTile(
+        val url: String,
+        val title: String,
+        val imagePath: String?,
+        // unique id used to identify specific home tiles, e.g. for deletion, etc.
+        val id: String
+) {
+
+    fun toJSONObject() = JSONObject(mapOf(
+                KEY_URL to url,
+                KEY_TITLE to title,
+                KEY_IMG to imagePath,
+                KEY_ID to id
+    ))
+
+    companion object {
+        fun fromJSONObject(jsonObject: JSONObject): HomeTile {
+            return HomeTile(jsonObject.getString(KEY_URL),
+                    jsonObject.getString(KEY_TITLE),
+                    jsonObject.optString(KEY_IMG),
+                    jsonObject.getString(KEY_ID))
+        }
+    }
+}
+
+/**
+ * Static accessor of custom home tiles, that is backed by SharedPreferences.
+ * This *MUST* be initialized before being used.
+ *
+ * New sites are appended to the end of the list.
+ *
+ * This keeps a cached version of the custom home tiles that have been pinned,
+ * in order to be more performant when checking whether sites are pinned or not.
+ * In order to keep the cache consistent, should only be called from the UIThread.
+ */
+class CustomTilesAccessor(context: Context) {
```

Comment:
nit: I prefer `CustomTilesStore` or similar. `Accessor` feels ambiguous: what is it accessing? Is this a wrapper with getters/setters around some private APIs? Is this persistent storage?

To me, `Store` indicates persistent storage and tells me where we're reading and storing it to.

Reply:
Going to rename (back) to CustomTilesManager (lol) because this isn't really a Store either.